Not CATCH bug?

### **Before/읽어주세요**

Make sure that you have used `/trg debug` so that the more detailed error message will show up in the console.
신고전에 `/trg debug` 명령어를 사용해서 디버그 모드를 켜주세요. 그래야 콘솔창에 더 자세한 정보가 나옵니다.

---

**Describe the bug/버그 증상**
17번 라인에서 해당 파일이 없으면 java.lang.reflect.InvocationTargetException/java.io.FileNotFoundException 로 Catch 되어야 하는데 catch로 넘어가지 않은 상황입니다.

If the file does not exist on line 17, it should be caught as java.lang.reflect.InvocationTargetException/java.io.FileNotFoundException, but it is not passed to catch.

**Server information/서버 정보**
1. Paper 1.16.5, Java 11 OpenJDK
2. 3.3.0.6 - Beta

**Error message/에러 메시지**
If error message (in red color) shows up in Minecraft client or on the console, report it here
에러메시지 (빨간색) 가 나타났다면 클라이언트와 콘솔의 메시지를 올려주세요.
![image](https://user-images.githubusercontent.com/6559842/152640700-1090c923-ba0c-4e95-bbda-a147d3e2e2bd.png)


**Others/그 밖의 정보들**

[11:36:16 INFO]: Limu_lhw issued server command: /filetest //커멘드 실행 (Command Execute)
[11:36:16 WARN]: java.lang.reflect.InvocationTargetException //라인 17 Exception (Line 17 Exception)
[11:36:16 INFO]: 2 //라인 19
[11:36:16 INFO]: :/ //라인 28. finally 씬

> Could you show me the full error log with debug mode on?

[21:08:11 INFO]: Limu_lhw issued server command: /filetest
[21:08:11 WARN]: java.lang.reflect.InvocationTargetException
[21:08:11 WARN]:        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance
0(Native Method)
[21:08:11 WARN]:        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance
(NativeConstructorAccessorImpl.java:62)
[21:08:11 WARN]:        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInst
ance(DelegatingConstructorAccessorImpl.java:45)
[21:08:11 WARN]:        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490
)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.tools.ReflectionUtil.constructNew(Reflec
tionUtil.java:512)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.call
Function(Interpreter.java:1179)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.inte
rpret(Interpreter.java:993)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.star
t(Interpreter.java:554)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.star
t(Interpreter.java:516)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.star
t(Interpreter.java:516)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.star
t(Interpreter.java:270)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.star
t(Interpreter.java:516)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.star
t(Interpreter.java:243)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.star
tWithContextAndInterrupter(Interpreter.java:186)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.manager.trigger.Trigger.start(Trigg
er.java:270)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.manager.trigger.Trigger$1.call(Trig
ger.java:225)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.manager.trigger.Trigger$1.call(Trig
ger.java:221)
[21:08:11 WARN]:        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[21:08:11 WARN]:        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolEx
ecutor.java:1128)
[21:08:11 WARN]:        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolE
xecutor.java:628)
[21:08:11 WARN]:        at java.base/java.lang.Thread.run(Thread.java:829)
[21:08:11 WARN]: Caused by: java.io.FileNotFoundException: plugins\TriggerReactor\test.txt (지정된
파일을 찾을 수 없습니다)
[21:08:11 WARN]:        at java.base/java.io.FileInputStream.open0(Native Method)
[21:08:11 WARN]:        at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
[21:08:11 WARN]:        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
[21:08:11 WARN]:        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:112)
[21:08:11 WARN]:        at java.base/java.io.FileReader.<init>(FileReader.java:60)
[21:08:11 WARN]:        ... 21 more
[21:08:11 WARN]: java.lang.reflect.InvocationTargetException
[21:08:11 WARN]:        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance
0(Native Method)
[21:08:11 WARN]:        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance
(NativeConstructorAccessorImpl.java:62)
[21:08:11 WARN]:        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInst
ance(DelegatingConstructorAccessorImpl.java:45)
[21:08:11 WARN]:        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490
)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.tools.ReflectionUtil.constructNew(Reflec
tionUtil.java:512)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.call
Function(Interpreter.java:1179)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.inte
rpret(Interpreter.java:993)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.star
t(Interpreter.java:554)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.star
t(Interpreter.java:516)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.star
t(Interpreter.java:516)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.star
t(Interpreter.java:270)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.star
t(Interpreter.java:516)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.star
t(Interpreter.java:243)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.star
tWithContextAndInterrupter(Interpreter.java:186)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.manager.trigger.Trigger.start(Trigg
er.java:270)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.manager.trigger.Trigger$1.call(Trig
ger.java:225)
[21:08:11 WARN]:        at io.github.wysohn.triggerreactor.core.manager.trigger.Trigger$1.call(Trig
ger.java:221)
[21:08:11 WARN]:        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[21:08:11 WARN]:        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolEx
ecutor.java:1128)
[21:08:11 WARN]:        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolE
xecutor.java:628)
[21:08:11 WARN]:        at java.base/java.lang.Thread.run(Thread.java:829)
[21:08:11 WARN]: Caused by: java.lang.NullPointerException
[21:08:11 WARN]:        at java.base/java.io.Reader.<init>(Reader.java:167)
[21:08:11 WARN]:        at java.base/java.io.BufferedReader.<init>(BufferedReader.java:101)
[21:08:11 WARN]:        at java.base/java.io.BufferedReader.<init>(BufferedReader.java:116)
[21:08:11 WARN]:        ... 21 more

It is better to use a code-block for readability.
```exception
java.lang.reflect.InvocationTargetException
    at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
    at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
    at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
    at io.github.wysohn.triggerreactor.tools.ReflectionUtil.constructNew(ReflectionUtil.java:512)
    at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.callFunction(Interpreter.java:1179)
    at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.interpret(Interpreter.java:993)
    at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.start(Interpreter.java:554)
    at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.start(Interpreter.java:516)
    at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.start(Interpreter.java:516)
    at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.start(Interpreter.java:270)
    at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.start(Interpreter.java:516)
    at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.start(Interpreter.java:243)
    at io.github.wysohn.triggerreactor.core.script.interpreter.Interpreter.startWithContextAndInterrupter(Interpreter.java:186)
    at io.github.wysohn.triggerreactor.core.manager.trigger.Trigger.start(Trigger.java:270)
    at io.github.wysohn.triggerreactor.core.manager.trigger.Trigger$1.call(Trigger.java:225)
    at io.github.wysohn.triggerreactor.core.manager.trigger.Trigger$1.call(Trigger.java:221)
    at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.io.FileNotFoundException: plugins\TriggerReactor\test.txt (지정된파일을 찾을 수 없습니다)
    at java.base/java.io.FileInputStream.open0(Native Method)
    at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
    at java.base/java.io.FileInputStream.(FileInputStream.java:157)
    at java.base/java.io.FileInputStream.(FileInputStream.java:112)
    at java.base/java.io.FileReader.(FileReader.java:60)
    ... 21 more
```

--- a/core/src/main/java/io/github/wysohn/triggerreactor/tools/ReflectionUtil.java
+++ b/core/src/main/java/io/github/wysohn/triggerreactor/tools/ReflectionUtil.java
@@ -490,7 +490,7 @@ public class ReflectionUtil {
         return classes;
     }
 
-    public static Object constructNew(Class<?> clazz, Object... args) throws NoSuchMethodException, InstantiationException, IllegalArgumentException, IllegalAccessException {
+    public static Object constructNew(Class<?> clazz, Object... args) throws NoSuchMethodException, InstantiationException, IllegalArgumentException, IllegalAccessException, InvocationTargetException {
         if (args.length < 1) {
             return clazz.newInstance();
         } else {
@@ -508,13 +508,7 @@ public class ReflectionUtil {
                 args = mergeVarargs(args, target.getParameterTypes());
             }
 
-            try {
-                return target.newInstance(args);
-            } catch (InvocationTargetException e) {
-                e.printStackTrace();
-            }
-
-            return null;
+            return target.newInstance(args);
         }
     }
 

--- a/core/src/test/java/io/github/wysohn/triggerreactor/core/script/interpreter/TestInterpreter.java
+++ b/core/src/test/java/io/github/wysohn/triggerreactor/core/script/interpreter/TestInterpreter.java
@@ -771,20 +771,15 @@ public class TestInterpreter {
     }
 
     @Test
-    public void testTryCatchInvokedMethod() throws IOException, ParserException, LexerException, InterpreterException {
+    public void testTryCatchInvokedMethod() throws Exception {
         Charset charset = StandardCharsets.UTF_8;
         String text = "" +
                 "import java.io.FileReader;" +
-                "import java.lang.AssertionError;" +
                 "" +
                 "TRY;" +
                 "    FileReader(\"./no-exist-file.data\");" +
                 "    #VERIFY false;" +
                 "CATCH e;" +
-                "    IF e.getClass() == AssertionError;" +
-                "        #VERIFY false;" +
-                "    ENDIF;" +
-                "" +
                 "    #VERIFY true;" +
                 "ENDTRY";
 
@@ -794,16 +789,18 @@ public class TestInterpreter {
         Node root = parser.parse();
 
         Map<String, Executor> executorMap = new HashMap<>();
-        executorMap.put("VERIFY", (timing, vars, context, args) -> {
-            Assert.assertEquals(true, args[0]);
-            return null;
-        });
+
+        Executor executor = mock(Executor.class);
+        when(executor.execute(any(), anyMap(), any(), anyBoolean())).thenReturn(null);
+        executorMap.put("VERIFY", executor);
 
         Interpreter interpreter = new Interpreter(root);
         interpreter.setExecutorMap(executorMap);
         interpreter.setTaskSupervisor(mockTask);
 
         interpreter.startWithContext(null);
+
+        verify(executor, times(1)).execute(any(), anyMap(), any(), eq(true));
     }
 
     @Test

--- a/core/src/test/java/io/github/wysohn/triggerreactor/core/script/interpreter/TestInterpreter.java
+++ b/core/src/test/java/io/github/wysohn/triggerreactor/core/script/interpreter/TestInterpreter.java
@@ -771,6 +771,42 @@ public class TestInterpreter {
     }
 
     @Test
+    public void testTryCatchInvokedMethod() throws IOException, ParserException, LexerException, InterpreterException {
+        Charset charset = StandardCharsets.UTF_8;
+        String text = "" +
+                "import java.io.FileReader;" +
+                "import java.lang.AssertionError;" +
+                "" +
+                "TRY;" +
+                "    FileReader(\"./no-exist-file.data\");" +
+                "    #VERIFY false;" +
+                "CATCH e;" +
+                "    IF e.getClass() == AssertionError;" +
+                "        #VERIFY false;" +
+                "    ENDIF;" +
+                "" +
+                "    #VERIFY true;" +
+                "ENDTRY";
+
+        Lexer lexer = new Lexer(text, charset);
+        Parser parser = new Parser(lexer);
+
+        Node root = parser.parse();
+
+        Map<String, Executor> executorMap = new HashMap<>();
+        executorMap.put("VERIFY", (timing, vars, context, args) -> {
+            Assert.assertEquals(true, args[0]);
+            return null;
+        });
+
+        Interpreter interpreter = new Interpreter(root);
+        interpreter.setExecutorMap(executorMap);
+        interpreter.setTaskSupervisor(mockTask);
+
+        interpreter.startWithContext(null);
+    }
+
+    @Test
     public void testNegation() throws Exception {
         Charset charset = StandardCharsets.UTF_8;
         String text = ""

--- a/core/src/test/java/io/github/wysohn/triggerreactor/tools/ReflectionUtilTest.java
+++ b/core/src/test/java/io/github/wysohn/triggerreactor/tools/ReflectionUtilTest.java
@@ -123,7 +123,7 @@ public class ReflectionUtilTest {
     }
 
     @Test
-    public void constructNew1() throws NoSuchMethodException, IllegalAccessException, InstantiationException {
+    public void constructNew1() throws NoSuchMethodException, IllegalAccessException, InstantiationException, InvocationTargetException {
         assertEquals(new OtherValue(), ReflectionUtil.constructNew(OtherValue.class));
 
         assertEquals(new OtherValue(1, 1), ReflectionUtil.constructNew(OtherValue.class,
@@ -140,7 +140,7 @@ public class ReflectionUtilTest {
     }
 
     @Test
-    public void constructNew2() throws NoSuchMethodException, IllegalAccessException, InstantiationException {
+    public void constructNew2() throws NoSuchMethodException, IllegalAccessException, InstantiationException, InvocationTargetException {
         //varargs in constructors
         assertEquals(new OtherValue(1, 55.55),
                 ReflectionUtil.constructNew(OtherValue.class, 1, 2.5, 3.5, 6.5, 9.9));

